### PR TITLE
feat(icons): add section symbol

### PR DESCRIPTION
**Pull request by gurtt**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

Adds a section symbol icon (§).

### Icon use case 

This icon adds to the current set of of document formatting marks including `pilcrow` and `dot`. It is used:
* to denote sections of a document
* to make reference to sections of other documents, especially in legal texts
* to refer to the justice system in general

### Alternative icon designs 

section-alt

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [x] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [ ] I've based them on the following Lucide icons: 
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/section.svg [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M12 15c2.208 0 4 1.344 4 3s-1.792 3-4 3-4-1.344-4-3" />
+  <path d="M12 9C9.792 9 8 7.656 8 6s1.792-3 4-3 4 1.344 4 3" />
```
Comment (jguddas) [inline icons/section.svg:13]:
```suggestion
  <path d="M12 15a4 3 0 1 1-4 3" />
  <path d="M16 6a4 3 0 1 0-4 3" />
```

**Review by jguddas — CHANGES_REQUESTED** [review]

**Review by ericfennis — APPROVED** [review]
Nice work!

**Comment by karsa-mistmere** [comment]
I'm not super thrilled about the curvature at these parts, I feel like it should be more fluid. 🤔
image

File: icons/section.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "gurtt"
```
Comment (karsa-mistmere) [inline icons/section.json:4]:
```suggestion
    "gurtt",
    "karsa-mistmere"
```

File: icons/section.svg [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M12 15a4 3 0 1 1-4 3" />
+  <path d="M16 6a4 3 0 1 0-4 3" />
+  <ellipse cx="12" cy="12" rx="4" ry="3" />
```
Comment (karsa-mistmere) [inline icons/section.svg:14]:
```suggestion
  <path d="M16 5a4 3 0 0 0-8 0c0 4 8 3 8 7a4 3 0 0 1-8 0" />
  <path d="M8 19a4 3 0 0 0 8 0c0-4-8-3-8-7a4 3 0 0 1 8 0" />
```

**Review by karsa-mistmere — CHANGES_REQUESTED** [review]

**Review by jguddas — APPROVED** [review]

**Pull request closed without merging**